### PR TITLE
nvidia-settings: add libraries to runpath

### PR DESCRIPTION
**Pull request by useranon350**
## Description of changes

Fixes issues with failed runtime library loading when running nvidia-settings -E and nvidia-settings -g.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I tested nvidia-settings -E and nvidia-settings -G, and also confirmed that the normal nvidia-settings GUI is still working.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by useranon350** [comment]
Would it be preferable to add libXv to the runtime dependencies to get rid of the manual ${libXv}/lib? I didn't change it to maintain the minimal set of the changes needed to fix the runtime library errors.

**Comment by FliegendeWurst** [comment]
> Would it be preferable to add libXv to the runtime dependencies to get rid of the manual ${libXv}/lib?

It would be best if the program works without manually configuring the runpath or similar.

**Comment by useranon350** [comment]
> > Would it be preferable to add libXv to the runtime dependencies to get rid of the manual ${libXv}/lib?
> 
> It would be best if the program works without manually configuring the runpath or similar.

With my changes, it works for me as a user without any need for manual runtime configuration. I am referring to the pre-existing libXv rpath, which is set by directly adding the libXv folder to the rpath config in the package config rather than using the macro to calculate the runpath for runtime dependencies automatically.

File: pkgs/os-specific/linux/nvidia-x11/settings.nix [hunk 1]
```diff
@@ -64,6 +65,12 @@ let
     '';
   };
 
+  runtimeDependencies = [
+    libglvnd libXrandr
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/nvidia-x11/settings.nix:69]: In that case, add libXv here.

And I think these are picked up by autoPatchelfHook if you add it.

**Pull request merged**